integration-tests/__tests__/deprecated-cli-options.test.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const path = require('path');
+const runJest = require('../runJest');
+
+const dir = path.resolve(__dirname, '../deprecated-cli-options');
+
+jest.mock('jest-config', () => ({
+  deprecationEntries: {
+    cache: () => 'This option is deprecated.',
+  },
+}));
+
+it('Prints deprecation warnings for CLI flags', () => {
+  const {stderr, stdout} = runJest(dir, ['--cache']);
```

Comment:
It's an end to end test, so we should assert on real values. Just pass `--preprocessorIgnorePatterns` which is already deprecated and you can get rid of mocking `jest-config`

Reply:
But the issue is `preprocessorIgnorePatterns` is not a CLI arg. It throws that warning. 

And what happens in the future those deprecations are removed from the code. That's why I thought I should mock it.

Reply:
Maybe we add an internal/undocumented cli arg / congif like --printDepricated

Reply:
I'd prefer being able to mock it rather than adding a hidden options. That maybe the last resort.